client/sites/components/sites-dataviews/dataviews-fields/site-field.tsx

```diff
@@ -60,7 +63,49 @@ const ListTileTitle = styled.div`
 	align-items: center;
 `;
 
-const SiteField = ( { site, openSitePreviewPane }: Props ) => {
+/**
+ * Renders an anchor element that can trigger analytics calls
+ * or client-side router navigation via the `onNavigate` prop.
+ */
+const Link = ( {
```

Comment:
My goal is to evolve this into a shared component to be used across the app. Keeping this here locally for now, until we validate the API with more use cases.